File: common/logisticspipes/api/ILPPipeConfigTool.java
package logisticspipes.api;

import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.item.ItemStack;

/**
 * public interface to be implemented by an item which can open the config GUI for a logistics pipe.
 * Some mod compatibility is already implemented inside LP.
 */
public interface ILPPipeConfigTool {

	default boolean canWrench(PlayerEntity player, ItemStack wrench, ILPPipeTile pipe) {
		return true;
	}

	default void wrenchUsed(PlayerEntity player, ItemStack wrench, ILPPipeTile pipe) {}

}

File: common/logisticspipes/api/ILPPipeConfigTool.kt
package logisticspipes.api

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack

/**
 * Implement this in an Item to allow pipe GUI configuration.
 *
 * Some mod compatibility is already implemented inside LP.
 */
interface ILPPipeConfigTool {
	/**
	 * @return true if the [player] is allowed to use the [tool][wrench] on a [pipe].
	 */
	fun canWrench(player: EntityPlayer, wrench: ItemStack, pipe: ILPPipeTile): Boolean

	/**
	 * TODO I wan unable to determine how this is used
	 */
	fun wrenchUsed(player: EntityPlayer, wrench: ItemStack, pipe: ILPPipeTile)
}
